core/src/test/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -321,14 +321,20 @@ public void deleteBlob(String blobName) throws IOException {
 
             @Override
             public void move(String sourceBlob, String targetBlob) throws IOException {
+                // simulate a non-atomic move, since many blob container implementations
+                // will not have an atomic move, and we should be able to handle that
                 maybeIOExceptionOrBlock(targetBlob);
-                super.move(sourceBlob, targetBlob);
+                super.writeBlob(targetBlob, super.readBlob(sourceBlob), 0L);
+                super.deleteBlob(sourceBlob);
```

Comment:
I pushed adb7aaded4ee9cfe19dd59781b3d85e570df9a57 to randomize between atomic and non-atomic